File: Structures_python/4_basic_data_structures/queue/queue.py
class Queue:
    def __init__(self):
        self.items = []

    def isEmpty(self):
        return len(self.items) == 0

    def enqueue(self, item):
        self.items.insert(0, item)

    def dequeue(self):
        return self.items.pop()

    def size(self):
        return len(self.items)


q = Queue()

q.enqueue('first')
q.enqueue('dog')
q.enqueue(True)

print(q.size())
print(q.dequeue())
print(q.dequeue())
print(q.size())

File: algorithm/js/queue.js
class Queue {
  constructor() {
    this._items = [];
  }

  //入队
  enqueue(...items) {
    this._items.push(...items);
    return this._items;
  }

  // 出队
  dequeue(count = 1) {
    //pull out the first item from the queue
    this._items.splice(0, count);
    return this._items;
  }

  peek() {
    //peek at the first item from the queue
    return this._items[0];
  }

  size() {
    //get the length of queue
    return this._items.length;
  }

  isEmpty() {
    //find whether the queue is empty or no
    return this._items.length === 0;
  }
}

// ------- 实例 -----
// 基数排序
// TODO: https://juejin.im/entry/59c9c3c86fb9a00a4f1b3e51
